app/divisions/[slug]/page.tsx

```diff
@@ -21,6 +22,9 @@ export async function generateMetadata({
 
 export default async function Division({ params, searchParams }) {
   const data = await getDivisionData(params.slug, searchParams.page);
+  if (data?.headers?.code === "404") {
```

Comment:
We talked about this but to be clear, this is because Repo API's 200 response returns the 404 _within_ the data content, correct? Can you add this info as a comment above this? It's not exactly a standard API response so it's worth mentioning it here.